packages/rollup-contracts/contracts/ovm/PartialStateManager.sol

```diff
@@ -0,0 +1,329 @@
+     * @param _ovmContractAddress The address of the OVM contract we'd like to associate with some code.
+     * @param _codeContractAddress The address of the code contract that's been deployed.
+     */
+    function associateCodeContract(address _ovmContractAddress, address _codeContractAddress) public {
+        require(msg.sender == address(executionManager));
+
+        ovmCodeContracts[_ovmContractAddress] = _codeContractAddress;
+    }
+
+    /**
+     * @notice Lookup the code contract for some OVM contract, allowing CALL opcodes to be performed.
+     * @param _ovmContractAddress The address of the OVM contract.
+     * @return The associated code contract address.
+     */
+    function getCodeContractAddress(address _ovmContractAddress) public returns(address) {
+        ensureVerifiedContract(_ovmContractAddress);
+
+        return ovmCodeContracts[_ovmContractAddress];
+    }
+
+    /**
+     * @notice Get the bytecode at some contract address. NOTE: This is code taken from Solidity docs here:
+     *         https://solidity.readthedocs.io/en/v0.5.0/assembly.html#example
+     * @param _codeContractAddress The address of the code contract.
+     * @return The bytecode at this address.
+     */
+    function getCodeContractBytecode(address _codeContractAddress) public view returns (bytes memory codeContractBytecode) {
+        assembly {
+            // retrieve the size of the code
+            let size := extcodesize(_codeContractAddress)
+            // allocate output byte array - this could also be done without assembly
+            // by using codeContractBytecode = new bytes(size)
+            codeContractBytecode := mload(0x40)
+            // new "memory end" including padding
+            mstore(0x40, add(codeContractBytecode, and(add(add(size, 0x20), 0x1f), not(0x1f))))
+            // store length in memory
+            mstore(codeContractBytecode, size)
+            // actually retrieve the code, this needs assembly
+            extcodecopy(_codeContractAddress, add(codeContractBytecode, 0x20), 0, size)
+        }
+    }
+
+    /**
+     * @notice Get the hash of the deployed bytecode of some code contract.
+     * @param _codeContractAddress The address of the code contract.
+     * @return The hash of the bytecode at this address.
+     */
+    function getCodeContractHash(address _codeContractAddress) public view returns (bytes32 _codeContractHash) {
+        // TODO: Look up cached hash values eventually to avoid having to load all of this bytecode
+        bytes memory codeContractBytecode = getCodeContractBytecode(_codeContractAddress);
+        _codeContractHash = keccak256(codeContractBytecode);
+        return _codeContractHash;
+    }
+
+    /**
+     * @notice Deploys a code contract, and then registers it to the state
+     * @param _ovmContractInitcode The bytecode of the contract to be deployed
+     * @return the codeContractAddress.
+     */
+    function deployContract(
```

Comment:
This will be changed what with moving safety check into EM, etc. so not reviewing for now